crates/ty_python_semantic/src/types.rs

```diff
@@ -3423,6 +3484,9 @@ impl<'db> Type<'db> {
             }
 
             Type::NominalInstance(instance) => instance.is_single_valued(db),
+            Type::NewTypeInstance(newtype) => {
+                Type::instance(db, newtype.base_class_type(db)).is_single_valued(db)
+            }
```

Comment:
same here -- this looks correct, but it would be great to have an explicit test for this